crypto/compactcert/verifier.go

```diff
@@ -50,13 +50,16 @@ func (v *Verifier) Verify(c *Cert) error {
 	sigs := make(map[uint64]crypto.Hashable)
 	parts := make(map[uint64]crypto.Hashable)
 	for pos, r := range c.Reveals {
-		sigs[pos] = r.SigSlot
+		sig, err := buildCommittableSignature(r.SigSlot)
+		if err != nil {
+			return err
+		}
+
+		sigs[pos] = sig
 		parts[pos] = r.Part
 
-		err := r.Part.PK.Verify(
-			uint64(r.Part.FirstValid),
+		err = r.Part.PK.Verify(
```

Comment:
There are multiple Verify calls in this function. 
Can you pleas add comments describing why each one is needed and what it is verifying?